lib/elixir/src/elixir_parser.yrl

```diff
@@ -256,6 +258,13 @@ access_expr -> bit_string : '$1'.
 access_expr -> sigil : build_sigil('$1').
 access_expr -> max_expr : '$1'.
 
+%% Numbers in different bases
+number -> binary : handle_literal(?exprs('$1'), '$1', {base, 2}).
+number -> octal : handle_literal(?exprs('$1'), '$1', {base, 8}).
+number -> decimal : handle_literal(?exprs('$1'), '$1', {base, 10}).
+number -> float : handle_literal(?exprs('$1'), '$1').
+number -> hex : handle_literal(?exprs('$1'), '$1', {base, 16}).
```

Comment:
I didn't mention this because since `char` is a different token I assumed we used it differently, but I now remember I made a PR a long time ago to use it as `char` so that it gives a nice error message on syntax errors. So big 👍 on me to go with `format` and collapse `char` into `number`.